### PR TITLE
Enable Windows build using Visual Studio 2015

### PR DESCRIPTION
**Pull request by peter-b**
This PR builds on #5134 and adds some additional patches to the build infrastructure to enable building on an actual Windows host.

N.b. this must be merged with livecode/livecode-thirdparty#91.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -504,10 +509,12 @@ def configure_android(opts):
     exec_gyp(args + opts['GYP_OPTIONS'])
 
 def configure_win(opts):
+    validate_target_arch(opts)
     validate_windows_tools(opts)
 
     args = core_gyp_args(opts) + ['-Gmsvs_version=' + opts['WIN_MSVS_VERSION']]
-    args += gyp_define_args(opts, {'ms_speech_sdk4': 'MS_SPEECH_SDK4',
+    args += gyp_define_args(opts, {'target_arch':    'TARGET_ARCH',
+                                   'ms_speech_sdk4': 'MS_SPEECH_SDK4',
```
Comment (livecodeali) [inline config.py]: I think this line can be removed, as the dependency on SAPI4 is removed earlier in this PR

**Comment by peter-b** [comment]
@livecodeali Thanks, I've addressed that.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok e294f29

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok e294f29b69bf7e3eb239c874ab114c7bb8740fe0

**Comment by livecode-vulcan** [comment]
😞 test failure e294f29b69bf7e3eb239c874ab114c7bb8740fe0

* try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/973)
* try-community-mac: [failure](https://vulcan.livecode.com/builders/try-community-mac/builds/979)
* try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/971)
* try-community-emscripten: [failure](https://vulcan.livecode.com/builders/try-community-emscripten/builds/972)
* try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/979)
* try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/993)
* try-community-android-armv6: [failure](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/970)

**Comment by livecodeali** [comment]
Depends on https://github.com/livecode/livecode/pull/5206

**Comment by livecodeali** [comment]
@livecode-vulcan retry e294f29

**Comment by livecodeali** [comment]
@livecode-vulcan review ok fb5499e

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok fb5499e64cd921103c61bbfec94169a29dad4b22

**Comment by livecode-vulcan** [comment]
😎 test success fb5499e64cd921103c61bbfec94169a29dad4b22

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1052)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1055)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1047)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1053)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1057)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1069)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1047)

**Pull request merged**